spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/StreamTemplate.java

```diff
@@ -28,16 +30,27 @@
  * Implementation for {@link StreamOperations}.
  *
  * @author Ilayaperumal Gopinathan
+ * @author Mark Fisher
  */
 public class StreamTemplate implements StreamOperations {
 
 	private final RestTemplate restTemplate;
 
-	private final Map<String, UriTemplate> resources;
+	private final UriTemplate definitionsPath;
+
+	private final UriTemplate deploymentsPath;
 
 	StreamTemplate(RestTemplate restTemplate, Map<String, UriTemplate> resources) {
 		this.restTemplate = restTemplate;
-		this.resources = resources;
+		this.definitionsPath = resources.get("streams/definitions");
```

Comment:
Now that we move this to constructor level. It would be nice to have some assertions to make sure the resources exist with these paths.